l10n_es_facturae/wizard/create_facturae.py

```diff
@@ -48,43 +32,52 @@ class CreateFacturae(models.TransientModel):
         default=True,
     )
 
+    @api.model
+    def create(self, vals):
+        if not vals.get("move_id", False):
+            move = self._get_move_from_context()
+            vals.update(
+                {
+                    "move_id": move.id,
+                }
+            )
+        res = super().create(vals)
+        return res
+
     def create_facturae_file(self):
-        log = Log()
-        move_ids = self.env.context.get("active_ids", [])
-        if not move_ids or len(move_ids) > 1:
-            raise UserError(_("You can only select one move to export"))
-        active_model = self.env.context.get("active_model", False)
-        assert active_model == "account.move", "Bad context propagation"
-        move = self.env["account.move"].browse(move_ids[0]).ensure_one()
-        if self.firmar_facturae:
+        # Validating mandatory fields
+        self.move_id.validate_facturae_fields()
+        skip_signature = self.env.context.get("skip_signature", False)
+        if self.firmar_facturae and not skip_signature:
             move_file = self.env.ref("l10n_es_facturae.report_facturae_signed")._render(
-                move.ids
+                self.move_id.ids
             )[0]
-            file_name = ("facturae_" + move.name + ".xsig").replace("/", "-")
+            file_name = ("facturae_" + self.move_id.name + ".xsig").replace("/", "-")
         else:
             move_file = self.env.ref("l10n_es_facturae.report_facturae")._render(
-                move.ids
+                self.move_id.ids
             )[0]
-            file_name = ("facturae_" + move.name + ".xml").replace("/", "-")
+            file_name = ("facturae_" + self.move_id.name + ".xml").replace("/", "-")
         file = base64.b64encode(move_file)
         self.env["ir.attachment"].create(
             {
                 "name": file_name,
                 "datas": file,
                 "res_model": "account.move",
-                "res_id": move.id,
+                "res_id": self.move_id.id,
                 "mimetype": "application/xml",
             }
         )
-        log.add(_("Export successful\n\nSummary:\nMove number: %s\n") % move.name)
         self.write(
             {
-                "note": log(),
                 "facturae": file,
                 "facturae_fname": file_name,
                 "state": "second",
             }
         )
+        self.move_id.write(
+            {"l10n_es_facturae_attachment": file, "facturae_fname": file_name}
```

Comment:
No has definido el campo. En cualquier caso, es más recomandable añadirlo como fichero para evitar crear un campo

Reply:
fichero > adjunto

Reply:
Un apunte, te he creado `2 pr `pero es cierto que tienen cierta relación entre ellos (con el campo `l10n_es_facturae_attachment` que comentas).
La forma correcta de subirlo sería hacer en un mismo pr `2 commits` distintos?
Entiendo que para hacerlo tendré que cerrar estos `2 pull request` y volver a hacer uno de nuevo.
Ojalá me puedas ayudar para que pueda aprender en un futuro.
Gracias!

Reply:
¿Necesitas por algo ese campo específico? Nosotros hablábamos de utilizar los adjuntos normales de Odoo (attachment_ids) en el chatter.

Reply:
En principio se podría hacer con lo que comentáis.

El apunte que he comentado yo es para saber cómo hay que subir los pr.

Reply:
Bueno, si los cambios tienen dependencia, tendrán que ir en el mismo PR, pero pueden ser distintos commits. La granularidad escogida es la suficiente como para poder entender cada cambio en su conjunto. El difícil equilibrio... jeje

Reply:
Entiendo. Va bien entonces si cierro estos 2 pr y hago uno nuevo con los cambios que comentáis de los attachments en 2 commits distintos? O como se tendría que proceder?

Reply:
Vale, adelante

Reply:
Ahora que lo he analizado mejor, el módulo como está ahora (una vez hecho este PR) ya se añaden como adjunton los ficheros. El campo l10n_es_facturae_attachment se utiliza a la vista para que el usuario obtenga únicamente el último fichero generado y pueda acceder fácilmente a él, en lugar de tener que buscar en los attachments cuál ha sido el último generado (que probablemente sea el primero que te aparece, pero igual no todos los usuarios lo ven de esa forma).

No creo que sea una mala idea tampoco esta implementación.

Ya me decís! Saludos.

Reply:
Yo lo veo redundante, la verdad.

Reply:
Vale, lo cambio pues